api/v1alpha1/clusterextension_types.go

```diff
@@ -319,7 +319,7 @@ type PreflightConfig struct {
 	// consequences of upgrading a CRD, such as data loss.
 	//
 	// This field is required if the spec.preflight field is specified.
-	CRDUpgradeSafety *CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety"`
+	CRDUpgradeSafety CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
```

Comment:
This field is a required field and therefore should not have the `omitempty` in the JSON tag:
```suggestion
	CRDUpgradeSafety CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety"`
```

Reply:
I originally left it without the `omitempty` tag, however, unit tests are failing with the below error i.e. the field having a null value when it is empty:
`        	Error Trace:	/Users/rashmigottipati/go/src/github.com/operator-framework/operator-controller/internal/controllers/clusterextension_controller_test.go:63
        	Error:      	Received unexpected error:
        	            	ClusterExtension.olm.operatorframework.io "cluster-extension-test-s58x6kfr" is invalid: [spec.preflight.crdUpgradeSafety.policy: Unsupported value: "": supported values: "Enabled", "Disabled", <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]`

 Since these fields are required only if the `spec.preflight` field is specified, there's a possibility of it being null when the `spec.preflight` field is not being set and so I think it would be ok to add the `omitempty` tag to this field and also the `Policy` field  but open to suggestions on how to appropriately handle this in a better way.

Reply:
Is there a way we could ensure the default value is set when the top level `spec.preflight` field is not explicitly set?

Reply:
if not, we may need to have some more complex CEL validations in place. We need to be careful there as well though to ensure the cost of the CEL expression is not too expensive.

Reply:
I found https://github.com/kubernetes-sigs/controller-tools/issues/622#issuecomment-1690128862 which may be what we need to do in this case.

Reply:
Keep in mind. This is required for now, only because there are no other sibling preflight check configurations. As soon as there are sibling configurations (e.g. for a permissions preflight check), then we'll a validation like "at least one of the preflight check fields is required".

Reply:
Thats a good point @joelanford . I do think it makes sense to go ahead and update it such that we have the "at least one of" approach